src/partition/src/row_splitter.rs

```diff
@@ -0,0 +1,323 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::iter;
+
+use api::helper;
+use api::v1::{ColumnSchema, Row, RowInsertRequest, Rows};
+use datatypes::value::Value;
+use store_api::storage::RegionNumber;
+
+use crate::error::Result;
+use crate::PartitionRuleRef;
+
+pub type RowInsertRequestSplits = HashMap<RegionNumber, RowInsertRequest>;
+
+pub struct RowSplitter {
+    partition_rule: PartitionRuleRef,
+}
+
+impl RowSplitter {
+    pub fn new(partition_rule: PartitionRuleRef) -> Self {
+        Self { partition_rule }
+    }
+
+    pub fn split(&self, req: RowInsertRequest) -> Result<RowInsertRequestSplits> {
+        // No partition
+        let partition_columns = self.partition_rule.partition_columns();
+        if partition_columns.is_empty() {
+            return Ok(iter::once((0, req)).collect());
```

Comment:
non-blocking: You could also use from.
```suggestion
            return Ok(HashMap::from([(0, req)]));
```